src/ruby/puppet-server-lib/puppet/server/network/http/handler.rb

```diff
@@ -32,7 +34,12 @@ def path(request)
   end
 
   def body(request)
-    request["body"]
+    body = request["body"]
+    if body.java_kind_of?(InputStream)
+      body.to_io.read()
```

Comment:
> I'm not sure there's anything which can be done to improve the memory utilization at this point.

Me either, was just wondering...  It does make sense that the behavior is the same as it always has been, so we're not making anything worse.  If it is actually a problem then the fix would have to be in Puppet anyway, not Puppet Server.